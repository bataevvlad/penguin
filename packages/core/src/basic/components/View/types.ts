import {ViewProps as DefaultViewProps, View as DefaultView} from 'react-native';

export type IViewProps = DefaultViewProps;
export type IView = DefaultView;
