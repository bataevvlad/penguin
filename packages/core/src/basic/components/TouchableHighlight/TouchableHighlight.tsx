import {forwardRef} from 'react';
import {TouchableHighlight as DefaultTouchableHighlight} from 'react-native';

import {ITouchableHighlight, TouchableHighlightProps} from './types';

// eslint-disable-next-line react/display-name
const TouchableHighlight = forwardRef<
  ITouchableHighlight,
  TouchableHighlightProps
>((props, ref) => {
  return <DefaultTouchableHighlight {...props} ref={ref} />;
});
export default TouchableHighlight;
