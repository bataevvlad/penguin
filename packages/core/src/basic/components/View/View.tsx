import {forwardRef} from 'react';
import {View as DefaultView} from 'react-native';

import {IView, IViewProps} from './types';

// eslint-disable-next-line react/display-name
const View = forwardRef<IView, IViewProps>((props, ref) => (
  <DefaultView ref={ref} {...props} />
));

export default View;
