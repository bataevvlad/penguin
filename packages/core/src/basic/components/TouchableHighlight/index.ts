import TouchableHighlight from './TouchableHighlight';

export * from './types';

export {TouchableHighlight};
