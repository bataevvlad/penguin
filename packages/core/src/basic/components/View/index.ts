import View from './View';

export * from './types';

export {View};
