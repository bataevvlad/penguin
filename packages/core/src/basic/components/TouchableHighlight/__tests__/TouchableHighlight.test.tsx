import {render} from '@testing-library/react-native';

import TouchableHighlight from '../TouchableHighlight';
import {View} from '../../View';

describe('@penguin/core/TouchableHighlight', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <TouchableHighlight underlayColor={'GRAY'} onPress={jest.fn}>
        <View />
      </TouchableHighlight>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
