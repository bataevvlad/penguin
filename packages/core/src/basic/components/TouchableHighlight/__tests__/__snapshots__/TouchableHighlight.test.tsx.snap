// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/TouchableHighlight renders correctly 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View />
</View>
`;
